src/tools/cephfs_mirror/Mirror.cc

```diff
@@ -496,31 +494,33 @@ void Mirror::update_fs_mirrors() {
   dout(20) << dendl;
 
   auto now = ceph_clock_now();
+
   double blocklist_interval = g_ceph_context->_conf.get_val<std::chrono::seconds>
     ("cephfs_mirror_restart_mirror_on_blocklist_interval").count();
-  bool check_blocklist = blocklist_interval > 0 && ((now - m_last_blocklist_check) >= blocklist_interval);
-
   double failed_interval = g_ceph_context->_conf.get_val<std::chrono::seconds>
     ("cephfs_mirror_restart_mirror_on_failure_interval").count();
-  bool check_failure = failed_interval > 0 && ((now - m_last_failure_check) >= failed_interval);
 
   {
     std::scoped_lock locker(m_lock);
     for (auto &[filesystem, mirror_action] : m_mirror_actions) {
       auto failed = mirror_action.fs_mirror && mirror_action.fs_mirror->is_failed();
       auto blocklisted = mirror_action.fs_mirror && mirror_action.fs_mirror->is_blocklisted();
 
-      if (check_failure && !mirror_action.action_in_progress &&
-	  !_is_restarting(filesystem) && failed) {
+      if (failed &&
+	  !_is_restarting(filesystem) &&
+	  !mirror_action.action_in_progress &&
```

Comment:
nit: just wondering if extracting these common conditions between the two `if (...)` conditionals could be easier on the eyes

Reply:
ACK - @dparmar18 commented the same, so, its probably hard to read. Will update.

Reply:
I meant something like ...
```
    if (!_is_restarting(filesystem) && !mirror_action.action_in_progress) {
        if (failed && failed_interval > 0 &&
            (mirror_action.fs_mirror->get_failed_ts() - now) > failed_interval) {
        ...
        ...
        ...
        } else if (blocklisted && blocklist_interval > 0 && 
                   (mirror_action.fs_mirror->get_blocklisted_ts() - now) > blocklist_interval) {
        ...
        ...
        ...
        }
    }
```